### ISSUE TITLE
Refresh project after task execution based on task outputs

### ISSUE THREAD

**Issue by staffanf** [eclipse/buildship#414]
If I run my "precompile" task to generate sources from the Gradle tasks view, I would like Buildship to automatically refresh the required files based on task outputs that have changed.
This would ensure that Eclipse picks up the changes and compiles the code correctly.

**Comment by donat** [comment]
Thanks for the proposal, it makes sense. However, I think it's easier for the users if Buildship would simply refresh the entire project after the task execution. We already have a [ticket](https://github.com/eclipse/buildship/issues/339) for that. Feel free to add your comments on it.

**Comment by oehme** [comment]
>  I think it's easier for the users if Buildship would simply refresh the entire project after the task execution. 

I don't think that's true, because

- the user has to make a decision whether to refresh or not
- refreshing the whole project is a big performance hit

Gradle knows which files have changed during a build and we could report that back through the TAPI and refresh just those files. 

#339 is a nice first step, but we should eventually implement the smarter solution that @staffanf is proposing.

**Comment by donat** [comment]
> the user has to make a decision whether to refresh or not

Just for completeness, #339 already proposes that.

**Comment by oehme** [comment]
Yes, that's why I said it's not a great solution. The user shouldn't have to make a decision, Gradle already knows what to refresh.

**Comment by staffanf** [comment]
@donat @oehme In our current #339 would probably be good enough but it's not that hard to imagine a case were refreshing the entire project view is not something you want to do after doing a precompile of a single subproject.
But any improvement would be appreciated.
